azurerm/internal/services/monitor/monitor_action_rule_suppression_resource.go

```diff
@@ -0,0 +1,517 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/location"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/monitor/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/monitor/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMonitorActionRuleSuppression() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMonitorActionRuleSuppressionCreateUpdate,
+		Read:   resourceArmMonitorActionRuleSuppressionRead,
+		Update: resourceArmMonitorActionRuleSuppressionCreateUpdate,
+		Delete: resourceArmMonitorActionRuleSuppressionDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ActionRuleID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ActionRuleName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"suppression": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"recurrence_type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(alertsmanagement.Always),
+								string(alertsmanagement.Once),
+								string(alertsmanagement.Daily),
+								string(alertsmanagement.Weekly),
+								string(alertsmanagement.Monthly),
+							}, false),
+						},
+
+						"schedule": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"end_date": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validate.ActionRuleScheduleDate,
+									},
+
+									"end_time": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validate.ActionRuleScheduleTime,
+									},
```

Comment:
would it make sense to combine these into a single property in the standard RFC format used mos]t places in the provider and populate the individual properties from it?